test/trento/domain/sap_system/sap_system_test.exs

```diff
@@ -395,7 +489,162 @@ defmodule Trento.SapSystemTest do
+                         features: "MESSAGESERVER",
+                         host_id: ^host_id,
+                         health: :passing
+                       },
+                       %SapSystem.Instance{
+                         features: "ABAP"
+                       }
+                     ]
+                   }
+                 } = state
+        end
+      )
+    end
+
+    test "should add an application instance to a non registered sap system when the instance is ABAP without complete a sap system registration" do
+      sap_system_id = Faker.UUID.v4()
+      sid = Faker.StarWars.planet()
+      db_host = Faker.Internet.ip_v4_address()
+      tenant = Faker.Beer.style()
+      instance_hostname = Faker.Airports.iata()
+      http_port = 80
+      https_port = 443
+      start_priority = "0.9"
+      host_id = Faker.UUID.v4()
+
+      initial_events = [
+        build(
+          :database_registered_event,
+          sap_system_id: sap_system_id,
+          sid: sid
+        ),
+        build(
+          :database_instance_registered_event,
+          sap_system_id: sap_system_id,
+          sid: sid,
+          tenant: tenant
+        )
+      ]
+
+      assert_events_and_state(
+        initial_events,
+        RegisterApplicationInstance.new!(%{
+          sap_system_id: sap_system_id,
+          sid: sid,
+          db_host: db_host,
+          tenant: tenant,
+          instance_number: "00",
+          instance_hostname: instance_hostname,
+          features: "ABAP",
+          http_port: http_port,
+          https_port: https_port,
+          start_priority: start_priority,
+          host_id: host_id,
+          health: :passing
+        }),
+        [
+          %ApplicationInstanceRegistered{
+            sap_system_id: sap_system_id,
+            sid: sid,
+            instance_number: "00",
+            instance_hostname: instance_hostname,
+            features: "ABAP",
+            http_port: http_port,
+            https_port: https_port,
+            start_priority: start_priority,
+            host_id: host_id,
+            health: :passing
+          },
+          %SapSystemHealthChanged{health: :passing}
+        ],
+        fn state ->
+          assert %SapSystem{
+                   sid: nil,
+                   application: %SapSystem.Application{
+                     sid: ^sid,
+                     instances: [
+                       %SapSystem.Instance{
+                         sid: ^sid,
+                         instance_number: "00",
+                         features: "ABAP",
+                         host_id: ^host_id,
+                         health: :passing
+                       }
+                     ]
+                   }
+                 } = state
+        end
+      )
+    end
+
+    test "should app an application instance to a non registered sap system when the instance is MESSAGESERVER without complete a sap system registration" do
```

Comment:
`should add`, `sap` -> `SAP`, `without a completing a SAP`...

Reply:
done